### ISSUE TITLE
Label track deleted after Tempo (or Time Signature)change and closing project (without saving)

### ISSUE THREAD

**Issue by LinkeHand** [audacity/audacity#6453]
### Bug description

Make a tempo change in the time signature toolbar to a project with a label track.
Close the project without saving.
The label track is deleted.

The result is even worse than in issue #6166, where the label track is only not in sync.

### Steps to reproduce

1. Generate Track
2. Add Label Track
3. Save Project - Save to computer
4. Close and reopen the file
5. Change Tempo (or Time Signature) in the Time Signature Toolbar
6. Close file - without saving
7. Reopen file

### Expected behavior

Closing a project does not change the project file at all.

### Actual behavior

After Step 4:

![DTMF_after save](https://github.com/audacity/audacity/assets/104369709/889b111b-1ff2-481c-a48e-b380bba3200e)

After Step 5 - tempo changed:

![DMTF_after tempo change](https://github.com/audacity/audacity/assets/104369709/e0530f5d-33e3-4a29-be26-dc39e016416a)

After Step 6 - reopened project file which was previously only closed without saving:

![DMTF_after tempo change _close_reopen](https://github.com/audacity/audacity/assets/104369709/57614d79-3bab-4d5f-ba89-40e28f145d8e)

### Audacity Version

current master / alpha / nightly build

### Operating system

Windows

### Additional context

Tested with Audacity 3.6.0-alpha-20240514  (audacity-win-3.6.0-alpha-20240514+3af928c-x64-msvc2022.zip).

**Comment by petersampsonaudacity** [comment]
Testing on W11 with the 3,6.0 Beta build

I confirm this bug - and testing shows that it is a regression on 3.5.1

**Comment by saintmatthieu** [comment]
It looks the problem must also have affected note and time tracks.

**Comment by dozzzzer** [comment]
Tested

**Comment by saintmatthieu** [comment]
Thanks @petersampsonaudacity .
I see another problem:
1. new project,
2. generate audio,
3. save project
4. change tempo
5. close without saving
6. re-open
Expecting the project tempo to be the original one, but it's the new one.
This was already like that before the PR I just merged, though.

**Comment by saintmatthieu** [comment]
... and that applies to time signature too.

**Comment by petersampsonaudacity** [comment]
@saintmatthieu @dozzzzer @LWinterberg 

Does Matthieu's "another problem" above need a new issue logging for it?

**Comment by saintmatthieu** [comment]
Your project had tempo 120, then you changed it to 180 and closed without saving. When you re-opened the project, the tempo was still 180, which is wrong. Had it been 120 as expected, your label track might have been in sync, so it might be considered related.

**Comment by petersampsonaudacity** [comment]
> Your project had tempo 120, then you changed it to 180 and closed without saving. When you re-opened the project, the tempo was still 180, which is wrong.

Yes, but all I'm doing is following the STRs for this logged issue ...

**Comment by petersampsonaudacity** [comment]
Testing on W10 with @saintmatthieu 's latest branch build for this:  audacity-win-3.6.0-alpha-20240624+7a627c1-x64

### Step 7
This mow looks better, correct in fact to me.
![image](https://github.com/audacity/audacity/assets/50205984/6d901ba4-8c7c-48f9-a535-3971026b4f77)
What I am now seeing is the project in the state in which it was last saved at step 3
a) the label track is not lost
b) the audio and labels are in sync
c) The track is at the 120 Tempo that it was at Step 4 (last save) and not the 180 Tempo that I changed it **_temporarily_** to at Step 5

So this looks fixed to me in Matthieu's latest branch build,

**Comment by petersampsonaudacity** [comment]
@LinkeHand 

Sorry to hear that Bernd, hope you mend well and quickly.

Don't worry about verifying, I'm sure the Muse QA team will thoroughly test this before any fix is released,  I have seen some code reviews already in PR #6745 - _and I'll still be giving any fix some exercise ..._

Peter

**Comment by chinakov** [comment]
Tested
